writeengine/shared/we_fileop.cpp

```diff
@@ -1975,11 +1975,22 @@ void FileOp::initDbRootExtentMutexes( )
 {
     boost::mutex::scoped_lock lk(m_createDbRootMutexes);
 
-    if ( m_DbRootAddExtentMutexes.size() == 0 )
+    std::vector<uint16_t> rootIds;
+    Config::getRootIdList( rootIds );
+    if (m_DbRootAddExtentMutexes.size() != rootIds.size())
     {
-        std::vector<uint16_t> rootIds;
-        Config::getRootIdList( rootIds );
+        //Something in configuration changed the size
+        //reset and rebuild this map
+        std::map<int, boost::mutex*>::iterator k = m_DbRootAddExtentMutexes.begin();
+        while (k != m_DbRootAddExtentMutexes.end() )
+        {
+            delete k->second;
+            ++k;
+        }
+    }
```

Comment:
I think this will fail in an odd scenario like this.
starting with dbroots 1, 2.
drop dbroot 2
add dbroot 3

the size check will pass, but the map will be wrong.

Reply:
also need to clear the map after the loop

Reply:
This is valid point. I'll come up with better solution. Started with this so issue wasn't forgotten about. Somewhat low priority and low threat of happening.